### PR TITLE
Publish docker image with ychaos pre-installed

### PR DESCRIPTION
**Pull request by vijaybabu4589**
# Summary

@yahoo/ychaos-dev

Temporarily push ychaos beta version to docker hub

---

#### Prerequisites
- [x] I have read the contribution guidelines

#### Code Analysis
- [ ] Covered by Unittests
- [ ] Security warnings ignored (Why?!)

#### Project related
- [ ] Schema changed and is backward compatible.
- [ ] Introduces a new sub-command for ychaos CLI

### Autogenerated Files
- [ ] Auto generated schema

### Pre-Merge Checklist

- [x] All the build jobs are passing

---

### Checklist (Reviewer 1)

- [ ] Reviewed Source Code
- [ ] Reviewed Tests (If applicable)
- [ ] Reviewed Documentation (If applicable)

### Checklist (Reviewer 2)

- [ ] Reviewed Source Code
- [ ] Reviewed Tests (If applicable)
- [ ] Reviewed Documentation (If applicable)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+FROM python:3.6-alpine
+RUN apk --update add --virtual build-dependencies libffi-dev gcc musl-dev && pip install --upgrade pip && pip install -i https://test.pypi.org/simple/ ychaos[chaos]
```
Comment (vijaybabu4589) [inline Dockerfile:2]: Temporarily push ychaos beta version to docker hub, once verified then i will make changes to push the main version

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,2 @@
+FROM python:3.6-alpine
```
Comment (AlfinST) [inline Dockerfile:1]: Any particular reason for python 3.6?
other than it being the lowest version we support?

Comment (vijaybabu4589) [reply Dockerfile:1]: i thought the lowest version we support is 3.6.. do we support a lower version than that ?

File: Dockerfile (hunk 1 above)
Comment (AlfinST) [inline Dockerfile:2]: A while back when I was testing  I had to add `--extra-index-url https://pypi.org/simple/` also for it to install properly from test.pypi  else it threw conflict errors
```
pip install -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ ychaos
```

https://stackoverflow.com/questions/60868060/module-on-test-pypi-cant-install-dependencies-even-though-they-exist

File: Dockerfile (hunk 2 above)
Comment (AlfinST) [reply Dockerfile:1]: Yes, 3.6 is the minimum version we support

File: Dockerfile (hunk 1 above)
Comment (AlfinST) [reply Dockerfile:2]: @shashankrnr32 helped me out for this ☝️

**Review by sushilkar — DISMISSED** [review]

**Comment by shashankrnr32** [comment]
@vijaybabu4589 Couple of comments

1. Please create a new directory for all the docker files. I am assuming you will need another one for prod package
2. Please add reasoning in the PR for each dependency that is being added in the line `apk add`

**Comment by vijaybabu4589** [comment]
> @vijaybabu4589 Couple of comments
> 
>     1. Please create a new directory for all the docker files (DockerFile). I am assuming you will need another one for prod package
> 
>     2. Please add reasoning in the PR for each dependency that is being added in the line `apk add`
> 
> 
> PR looks great overall
1. Since we are only containerizing prod version we only need one docker file for now
2. `python-alpine` doesn't come with few c libs like gcc which are used while installing packages so manually install them. Issue: https://github.com/docker-library/python/issues/318

**Review by shashankrnr32 — APPROVED** [review]
Approved.

File: Dockerfile (hunk 1 above)
Comment (vijaybabu4589) [reply Dockerfile]: @AlfinST Any ways i have manually remove pushed images from docker repo before adding publish step to SD release flow so i removed beta version installation and installing prod version

File: screwdriver.yaml [hunk 3]
```diff
@@ -143,6 +143,17 @@ jobs:
         template: python/package_python
         requires: [version_test_pypi]
 
+    # publish latest ychaos version to docker hub
+    publish_test_docker:
```
Comment (vijaybabu4589) [inline screwdriver.yaml:147]: For testing purpose adding publish step here.. once verified i will move it `release` flow

**Review by AlfinST — APPROVED** [review]

**Review by shashankrnr32 — APPROVED** [review]

**Pull request merged**